### ISSUE TITLE
ElectronCash 2.92 client will not connect to any servers (I tried manually and SSL checked)

### ISSUE THREAD

**Issue by BCCDave** [fyookball/electrum#119]
I have tried every variation I can think of settings I can think of, but no matter what I do my status is always not connected.  

I started with 2.92 on the same Win 7 Virtualbox VM that I run electrum on.  Seemed to install fine, but could not connect, also does not show either fork on network tab.  I tried all of the BCH servers manually one after another, SSL & no SSL, no change.  Auto select or manual, no change.  

Then I tried deleting the blockchain_headers file.  It came back at the same size so after waiting to see if it would sync up, I change the electrum folder name and deleted the Electron Cash blockchain_headers file again.  It came back the same size (around 37mb) but still no connection.  

I ran ccleaner to delete all the tmp files, etc. and deleted the block headers file again, it came back, but still no connection according to client (although it sometimes seems to be able to get headers).

I tried re-downloading and reinstalling multiple times checking hashes every time, same story.  
Then I repeated all of the above inside multiple other VM's, Win 7x64, Win 10x64, Linux Mint 17.3x86, all with the same result.  I also tried ver 2.9.0 on a few machines, manually selecting the server, but still no luck.
My internet is over WiFi (not by choice) and I am running through a VPN, but Electrum is working fine, so I am at a loss.

Thanks in advance

**Comment by fyookball** [comment]
Sounds like it maybe didnt finish downloading the headers.  I will check about the speed of the bitcoincash.com server

**Comment by BCCDave** [comment]
I left some of these machines running for hours at a time.  The thing with
the header file was weird, because sometimes it would get that 37mb, but
sometimes nothing.  Also I should have mentioned, the certs folder was
always empty when I checked, I think the forks as was as well.

**Comment by molecular** [comment]
have you also deleted the "forks" folder along with blockchain_headers file?

Do you have the possibility to start from a terminal with the "-v" switch (something like (for win): start button, enter "cmd", drag&drop electron.exe onto terminal, add " -v" to the end of the exe, hit <ret>". Then try copy-pasting the output.

**Comment by BCCDave** [comment]
Forks folder is empty, Network tab shows no forks at all instead of two.  I have done clean installs of 2.92 in  half a dozen VM's where Electrum had never been installed, so forks folder, etc. would have been created from scratch.

I haven't been able to find a log for the Windows version, but I ran 2.92 &2.90 with -v flag in Linux Mint 17.3 and got this.

VirtualBox ~/Electron Cash-2.9.2 $ sudo electron-cash -v

Error: could not find paymentrequest_pb2.py. Create it with 'protoc --proto_path=lib/ --python_out=lib/ lib/paymentrequest.proto'

VirtualBox ~/Electron Cash-2.9.0 $ sudo electron-cash -v

Error: could not find paymentrequest_pb2.py. Create it with 'protoc --proto_path=lib/ --python_out=lib/ lib/paymentrequest.proto'

I ran, 
'protoc --proto_path=lib/ --python_out=lib/ lib/paymentrequest.proto
and got a missing file message which I seem to have misplaced.  I'm on another machine now, but will find it and add to this later.

I would think that the problem was running out of a VM, but Electrum runs fine out of VM's so I am at a loss.

**Comment by BCCDave** [comment]
Here is the whole sequence in Linux Mint 17.3 32bit VirtualBox machine.

$ sudo electron-cash -v
Error: could not find paymentrequest_pb2.py. Create it with 'protoc --proto_path=lib/ --python_out=lib/ lib/paymentrequest.proto'
$ protoc --proto_path=lib/ --python_out=lib/ lib/paymentrequest.proto
lib/paymentrequest.proto: No such file or directory

**Comment by fyookball** [comment]
we just publisehd 2.9.3

**Comment by BCCDave** [comment]
2.9.3 has the same problem, but I figured it out.  The issue is with the
VPN.  I turned off the VPN and was able to connect and make transfers with
no problem.

I probably won't use Bitcoin Cash again until the VPN issue with your
client is fixed, but thanks for your work and help, getting control of my
money was a big relief.  Just to reiterate what I said in the original
post, Electrum works fine with my machine & VPN, etc., so I am not sure
where the issue is since you share so much of the same code, but hopefully
this will help you find it.

Thanks again
